apps/web/src/lib/auth/getSession.ts

```diff
@@ -0,0 +1,46 @@
+import { lucia } from '$/lib/auth'
+import type { Session, User } from 'lucia'
+
+export async function getSession(): Promise<
+  { user: User; session: Session } | { user: null; session: null }
+> {
+  // NOTE: We dynamically import the cookies function to make Nextjs happy
+  // Info: https://github.com/vercel/next.js/issues/49757
+  const { cookies } = await import('next/headers')
+  const sessionId = cookies().get(lucia.sessionCookieName)?.value ?? null
+
+  if (!sessionId) {
+    return {
+      user: null,
+      session: null,
+    }
+  }
+
+  const result = await lucia.validateSession(sessionId)
+
+  try {
+    if (result.session && result.session.fresh) {
+      const sessionCookie = lucia.createSessionCookie(result.session.id)
+      cookies().set(
+        sessionCookie.name,
+        sessionCookie.value,
+        sessionCookie.attributes,
+      )
+    }
+
+    if (!result.session) {
+      const sessionCookie = lucia.createBlankSessionCookie()
+
+      cookies().set(
+        sessionCookie.name,
+        sessionCookie.value,
+        sessionCookie.attributes,
+      )
+    }
+  } catch {
+    // ignore
+    // next.js throws when you attempt to set cookie when rendering page
```

Comment:
so this is the recommended way of doing sessions in nextjs with lucia? looks hacky to me

Reply:
Yep https://github.com/latitude-dev/latitude-llm/pull/33#discussion_r1677487956